### DISCUSSION TITLE
The Future of Field Customizers

### DISCUSSION THREAD

**Discussion by AJIXuMuK** [SharePoint/sp-dev-docs#7407] [Ideas]
Hello folks!
Similarly to #7375, starting this discussion to collect your feedback on Field Customizers.
We're currently working on fixing some issues but also want to improve the adoption and usage of this type of extensions.

Please, share your thoughts, feedback and use cases!

I will start with few things that were mentioned multiple times:
- Field Customizers should work in Teams
- Ability to apply Field Customizers to New/Edit/View forms (panels)
- Field Customizers in Quick Edit View

Please, feel free to share your ideas and use cases!
And thank you for your time and input!

**Comment by andrewconnell** [comment]
In addition to the ones listed in the OP, all of which get +💯  from me (ESP the ones about having an edit experience which was stated a preview was coming within a few months after their initial release in May 2017) I'll add:

- should work & be rendered in all native Microsoft apps (SharePoint/Teams/etc) where lists are rendered (similar to the OP's points)
- should have an "edit" experience for new/edit forms, panels, etc
- ability to connect them to an existing site column
  - _today this is only possible with a manual post-install step by running some code against the SP REST API_

**Comment by patrikhellgren** [comment]
I fully agree with everything said and here are some additional thoughts:

- Field customizers should also be available in the filter panel
- It would be great if we had a way to replace the label and calculation of the column headers features Sort by, Group by and Filter by. We sometimes store for example a json object in a text field and have a field customizer to display a representation of that json object. In those cases it would be great if we could assign our own sort by, group by and filter by functions. In those cases I would also like to be able to change the sorting label accordingly, out of the box it says "Smaller to larger" for number fields, "A to Z" for text fields and "Older to newer" for date fields. If I have a json object in a text field it would always say "A to Z" even if I wanted a sort on a date in the json object.
- We often need more info about the list, page, user etc so something like the onGetState that was mentioned in #7375 would be nice.

**Comment by patmill** [comment]
Allow better access to URL field data - https://github.com/SharePoint/sp-dev-docs/issues/1167

**Comment by diabhoil** [comment]
This topic is kinda old. Is there some work in progress for field customizers? At the moment there is e.g. no way to change a single field in the default new/edit form with custom complex code. Would love to hear that MS is working on some updates...
